### PR TITLE
python3Packages.troposphere: init at 4.1.0

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Description of changes
Add [troposphere](https://github.com/cloudtools/troposphere) and [awacs](https://github.com/cloudtools/awacs), since troposphere depends on it. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/awacs/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+buildPythonPackage rec {
+  pname = "awacs";
+  version = "2.2.0";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-zWRQHxjHogmSKSqnvQLJCdoilkOlJ07p00lN+U46WkU=";
```
Comment (dotlambda) [inline pkgs/development/python-modules/awacs/default.nix:19]:
```suggestion
    hash = "sha256-zWRQHxjHogmSKSqnvQLJCdoilkOlJ07p00lN+U46WkU=";
```

File: pkgs/development/python-modules/awacs/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTests = [ "test_examples" ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/awacs/default.nix:30]: Why?

File: pkgs/development/python-modules/awacs/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  pythonImportsCheck = [ "awacs" ];
+
+  meta = with lib; {
+    description = "Amazon Web Access Control Subsystem";
+    maintainers = with maintainers; [ jlesquembre ];
+    license = with licenses; [ bsd2 ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/awacs/default.nix:37]:
```suggestion
    license = licenses.bsd2;
```

File: pkgs/development/python-modules/awacs/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTests = [ "test_examples" ];
+
+  pythonImportsCheck = [ "awacs" ];
+
+  meta = with lib; {
+    description = "Amazon Web Access Control Subsystem";
```
Comment (dotlambda) [inline pkgs/development/python-modules/awacs/default.nix:35]: I like something like
```suggestion
    description = "AWS Access Policy Language creation library";
```
better.

File: pkgs/development/python-modules/troposphere/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    cfn-flip
+    typing-extensions
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  # PyPi package does not include tests.
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:35]: You can use `fetchFromGitHub` instead.

File: pkgs/development/python-modules/troposphere/default.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  pythonImportsCheck = [ "troposphere" ];
+
+  meta = with lib; {
+    description = "library to create AWS CloudFormation descriptions";
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:39]:
```suggestion
    description = "Library to create AWS CloudFormation descriptions";
```

File: pkgs/development/python-modules/troposphere/default.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    maintainers = with maintainers; [ jlesquembre ];
+    license = with licenses; [ bsd2 ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:41]:
```suggestion
    license = licenses.bsd2;
```

File: pkgs/development/python-modules/troposphere/default.nix [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-7SA8Eu8P/b47pDiOyV25gxx25E5nP0yw96CRpb2Y1BY=";
+  };
+
+  propagatedBuildInputs = [
+    awacs
+    cfn-flip
+    typing-extensions
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:27]: only required for Python < 3.8: https://github.com/cloudtools/troposphere/blob/4.1.0/setup.cfg#L34

File: pkgs/development/python-modules/troposphere/default.nix [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-7SA8Eu8P/b47pDiOyV25gxx25E5nP0yw96CRpb2Y1BY=";
+  };
+
+  propagatedBuildInputs = [
+    awacs
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:25]: This should be in `checkInputs` and `passthru.optional-dependencies.policy`.

File: pkgs/development/python-modules/awacs/default.nix [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-zWRQHxjHogmSKSqnvQLJCdoilkOlJ07p00lN+U46WkU=";
+  };
+
+  propagatedBuildInputs = [
+    typing-extensions
```
Comment (dotlambda) [inline pkgs/development/python-modules/awacs/default.nix:23]: only required for Python < 3.8: https://github.com/cloudtools/awacs/blob/2.2.0/setup.cfg#L31

File: pkgs/development/python-modules/awacs/default.nix [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+  # python dependencies
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "awacs";
+  version = "2.2.0";
+
+  disabled = pythonOlder "3.6";
```
Comment (dotlambda) [inline pkgs/development/python-modules/awacs/default.nix:15]:
```suggestion
  disabled = pythonOlder "3.7";
```
https://github.com/cloudtools/awacs/blob/2.2.0/setup.cfg#L33

File: pkgs/development/python-modules/troposphere/default.nix [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+, awacs
+, cfn-flip
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "troposphere";
+  version = "4.1.0";
+
+  disabled = pythonOlder "3.6";
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:17]:
```suggestion
  disabled = pythonOlder "3.7";
```
https://github.com/cloudtools/troposphere/blob/4.1.0/setup.cfg#L43

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/awacs/default.nix (hunk 2 above)
Comment (jlesquembre) [reply pkgs/development/python-modules/awacs/default.nix:32]: I added a comment with the reason

**Comment by jlesquembre** [comment]
@dotlambda Thanks for the review, I updated the PR with your suggestions

File: pkgs/development/python-modules/troposphere/default.nix [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    pytestCheckHook
+    awacs
+  ];
+
+  passthru.optional-dependencies = {
+    policy = [ awacs ];
+  };
+
+  # https://github.com/cloudtools/troposphere/blob/4.1.0/Makefile#L87-L88
+  checkPhase = "python setup.py test";
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:42]:
```suggestion
```
No need for this if you use `pytestCheckHook`.

File: pkgs/development/python-modules/troposphere/default.nix [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+    repo = pname;
+    rev = version;
+    hash = "sha256-cAn4Hty5f/RsCnUA59CxtGrhRgzVyaHe5PuQOM6lwEQ=";
+  };
+
+  propagatedBuildInputs = [
+    cfn-flip
+  ] ++
+  lib.lists.optional (pythonOlder "3.8")
+    typing-extensions;
```
Comment (dotlambda) [inline pkgs/development/python-modules/troposphere/default.nix:30]:
```suggestion
  ] ++ lib.lists.optionals (pythonOlder "3.8") [
    typing-extensions
  ];
```

File: pkgs/development/python-modules/awacs/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/awacs/default.nix:32]: I suggest using
```nix
checkPhase = ''
  ${python.interpreter} -m unittest discover
''
```
No need for `pytestCheckHook`.

File: pkgs/development/python-modules/awacs/default.nix [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-zWRQHxjHogmSKSqnvQLJCdoilkOlJ07p00lN+U46WkU=";
+  };
+
+  propagatedBuildInputs =
+    lib.lists.optional (pythonOlder "3.8")
+      typing-extensions;
```
Comment (dotlambda) [inline pkgs/development/python-modules/awacs/default.nix:24]:
```suggestion
  propagatedBuildInputs = lib.lists.optionals (pythonOlder "3.8") [
    typing-extensions
  ];
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please split your changes into two commits.

File: pkgs/development/python-modules/troposphere/default.nix (hunk 13 above)
Comment (jlesquembre) [reply pkgs/development/python-modules/troposphere/default.nix]: Not exactly the same, `pytestCheckHook` runs `${python.interpreter} -m pytest`, which for some reason fails for one test, while `${python.interpreter} setup.py test` works. I removed `pytestCheckHook` and let the `${python.interpreter} setup.py test`

**Comment by jlesquembre** [comment]
Thanks again @dotlambda . I split the changes, and applied your suggestions, except this one: https://github.com/NixOS/nixpkgs/pull/201803#discussion_r1027146721 (since the tests fail with that change) Do you see a better alternative?

File: pkgs/development/python-modules/troposphere/default.nix (hunk 13 above)
Comment (dotlambda) [reply pkgs/development/python-modules/troposphere/default.nix]: Does `${python.interpreter} -m unittest discover` work? That's preferred over using `setup.py`.

Comment (jlesquembre) [reply pkgs/development/python-modules/troposphere/default.nix]: Yes, that works too, I just updated it

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 201803` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.awacs</li>
    <li>python310Packages.troposphere</li>
    <li>python39Packages.awacs</li>
    <li>python39Packages.troposphere</li>
  </ul>
</details>

**Pull request merged**